common/network-common/src/main/java/org/apache/spark/network/protocol/MessageWithHeader.java

```diff
@@ -44,6 +44,27 @@
   private final long bodyLength;
   private long totalBytesTransferred;
 
+  @Override
+  public FileRegion touch() {
+    return this;
+  }
+
+  @Override
+  public FileRegion touch(Object hint) {
+    return this;
+  }
+
+  @Override
+  public FileRegion retain() {
+    return this;
```

Comment:
I believe this breaks the netty reference counting logic, I think instead it should be
```
return (TestFileRegion)super.retain();
```